auction9-backend/handler.js

```diff
@@ -171,15 +171,82 @@ export const stopActiveAuction = async (event, context) => {
  * GET: /wonauctions
  */
 export const getUserWonAuctions = async (event, context) => {
-   try {
-     let currentUserId = event.multiValueQueryStringParameters.userId[0];
-     let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
-     await mysql.end();
-     return generateResponse(200, resultsUserWonAuctions);
-   } catch (error) {
-     console.log(error);
-     return generateResponse(400, {
-       message: 'There was an error while getting my won auctions.'
-     });
-   }
+  try {
+    let currentUserId = event.multiValueQueryStringParameters.userId[0];
+    let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
+    await mysql.end();
+    return generateResponse(200, resultsUserWonAuctions);
+  } catch (error) {
+    console.log(error);
+    return generateResponse(400, {
+      message: 'There was an error while getting my won auctions.'
+    });
+  }
+};
+
+
+/* Allowed status changes:
+ * order status: INACTIVE -> ACTIVE -> FINISHED -> REALIZED
+ * current -> new :
+ * INACTIVE -> ACTIVE   [x]
+ * ACTIVE -> INACTIVE   [x]
+ * ACTIVE -> FINISHED   [x]
+ * FINISHED -> REALIZED [x]
+ */
+/* realizeFinishedAuction - will update status for auction to 'realized'
+ * PUT: /myauctions/id
+ */
+export const realizeFinishedAuction = async (event, context) => {
+  try {
+    let reqBody = JSON.parse(event.body);
+    let auctionId = reqBody.auction.auctionID;
+    let currentStatus = await mysql.query(`SELECT status FROM tbl_auction WHERE auctionID=?`,  [auctionId]);
+    await mysql.end();
+    // requested status to be changed into
+    let reqStatus = reqBody.changeStatus;
+    const statuses = {
```

Comment:
You can move statuses enum from method, if it's on the handler level, all method can access and use the same one.

Reply:
Will be moved in next PR